numpy/meson.build

```diff
@@ -29,6 +30,9 @@ if is_windows
     endif
   endif
 endif
+if is_msvc
+      add_project_arguments('-DNDEBUG', language: ['c', 'cpp'])
+endif
```

Comment:
While maybe not strictly needed, it is conventional to add this to a release build

Reply:
But this is not for a release build, this is unconditional? There's some discussion at https://stackoverflow.com/questions/2290509/debug-vs-ndebug, and it may be related to gh-23122. Not sure what is right here, but a comment about why this is here would be great if we need to keep it.

Reply:
Maybe we could leave supporting a debug build for a future PR. I think meson should be adding that flag in general for release builds. As it is, it took quite an effort to work out the build command sequence
```
meson setup build --prefix=$PWD\build-install -Ddebug=false --optimization 2 --vsenv
meson compile -C build -v
cd build
meson install --no-rebuild
cd ..
```

I will add a comment

Reply:
Hmm, removing the define, now the [build fails](https://github.com/numpy/numpy/actions/runs/4230989854/jobs/7348965062#step:7:299) at an assert
```
assert(ts->field_buffer_length >= 2*(ssize_t)sizeof(Py_UCS4));
```

since `ssize_t` is not defined for windows at that point ...

Reply:
fixed. Compilation without NDEBUG succeeds on windows.